pkg/backend/api/api_grpc_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	accountclientmock "github.com/bucketeer-io/bucketeer/pkg/account/client/mock"
+	"github.com/bucketeer-io/bucketeer/pkg/cache"
+	cachev3mock "github.com/bucketeer-io/bucketeer/pkg/cache/v3/mock"
+	featureclientmock "github.com/bucketeer-io/bucketeer/pkg/feature/client/mock"
+	"github.com/bucketeer-io/bucketeer/pkg/log"
+	accountproto "github.com/bucketeer-io/bucketeer/proto/account"
+	environmentproto "github.com/bucketeer-io/bucketeer/proto/environment"
+)
+
+func TestCheckAuth(t *testing.T) {
+	t.Parallel()
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	patterns := []struct {
```

Comment:
There are many missing cases for the `CheckAuth`.

- ErrContextCanceled
- ErrInternal
- ErrMissingAPIKey
- ErrDisabledAPIKey
- ErrDisabledAPIKey

Reply:
tests added
https://github.com/bucketeer-io/bucketeer/pull/1012/commits/572f9b675bd52803221ff8097f2acbcb6420fdd3